### PR TITLE
Map block: fix Mapkit marker captions

### LINKED ISSUE #30457: Map block: map marker doesn't show the label entered
When a user sets a marker's label, this doesn't get displayed in the frontend: 

Image

### PR DESCRIPTION
**Pull request by TimBroddin**
Fixes #30457

## Proposed changes:

This PR shows the caption of markers on a Mapkit map when clicking them.

| Before | After |
|-|-|
| CleanShot 2023-05-04 at 15 39 04@2x | CleanShot 2023-05-04 at 15 38 51@2x |

Since Apple doesn't provide a way to only change the text contents of the marker callout, we lost the marker icon. I think it's cleaner this way.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
1. Apply this PR (either by running Jetpack locally or using the JP downloader to your sandbox)
2. If running locally enable Mapkit by setting the cookie (`document.cookie = 'map_provider=mapkit';`)
3. Add a map with a marker:

CleanShot 2023-05-04 at 15 52 15@2x

4. View this map on the frontend

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/map/view/mapkit.js [hunk 1]
```diff
@@ -99,13 +99,32 @@ class MapkitBlock {
...
+						title.style.fontSize = '17px';
+						title.style.fontWeight = '500';
+						title.style.lineHeight = '17px';
+						title.style.marginTop = '4px';
+						title.textContent = annotation.title;
+
+						const caption = element.appendChild( document.createElement( 'p' ) );
+						caption.style.fontSize = '14px';
+						caption.style.margin = '0px 0px 4px 0px';
+						caption.textContent = annotation.data?.caption;
```
Comment (yansern) [inline projects/plugins/jetpack/extensions/blocks/map/view/mapkit.js:120]: To combat long captions, setting `maxWidth` to `50%` doesn't work, but setting `width` to `(annotation.element.parentElement.offsetWidth / 2) + "px"` does.

**Before:**
image

**After:**
image

**Review by yansern — DISMISSED** [review]

**Comment by yansern** [comment]
Both sidebar and inline captions work.

image


image

**Comment by yansern** [comment]
I feel that marker title could use some font-weight or whitespace tweaking, that's how it looks like on my end.
image

**Comment by yansern** [comment]
Perhaps the input could be debounced? 

https://github.com/Automattic/jetpack/assets/1287077/0913ae90-93cd-4423-b128-c81634fac40f

**Comment by yansern** [comment]
All the above aren't blockers for this PR but would be nice for a nicer experience.

File: projects/plugins/jetpack/extensions/blocks/map/view/mapkit.js (hunk 1 above)
Comment (TimBroddin) [reply projects/plugins/jetpack/extensions/blocks/map/view/mapkit.js:120]: Thanks!

**Comment by TimBroddin** [comment]
> Perhaps the input could be debounced?

I don't think debouncing is the main problem, but the "new marker" animation gets played on every change. Will fix this in https://github.com/Automattic/jetpack/pull/30482 since that one has updated marker code.

**Comment by TimBroddin** [comment]
The other PR doesn't have this issue: 

CleanShot 2023-05-10 at 13 30 40

**Comment by yansern** [comment]
Retested. Works good.
image

**Review by yansern — APPROVED** [review]

**Review by zinigor — APPROVED** [review]
Tested with a long caption, works well for me!

**Pull request merged**